server/controllers/moleculesImporter.js

```diff
@@ -0,0 +1,262 @@
+            res.sendServerError(addErrorTitle(error, "Can't insert new molecules"));
+            deleteUploadedFile();
+          });
+      } else {
+        const warnings = analyzeData(data);
+        res.sendResponse(200, {
+          message: "File tested but not imported",
+          warnings,
+          imported,
+        });
+        deleteUploadedFile();
+      }
+    })
+    .catch((error) => {
+      if (HeaderErrors.isInstance(error)) {
+        return res.sendUsageError(400, "Bad formatted file", {
+          errors: error.errors,
+          imported,
+        });
+      }
+      deleteUploadedFile();
+      return res.sendServerError(error);
+    });
+}
+
+/**
+ *
+ * @api {get} /import/molecules Get the last imported file
+ * @apiName GetLastImported
+ * @apiGroup Import
+ * @apiPermissions LoggedIn 
+ * @apiPermissions Admin 
+ *
+ * @apiSuccess (200) {string} file The url to the file
+ *
+ *
+ * @apiSuccessExample Success-Response:
+ *  {
+      "file": "https://glowing-octo-guacamole.com/api/v1/molecules/molecules_1612279095021.csv"
+    }
+ *
+ *
+ */
+function getLastImportedFile(req, _res) {
+  const res = new HttpResponseWrapper(_res);
+  getFiles(path.resolve("files", "molecules"))
+    .then((files) => {
+      const last = files.reduce(
+        ({ filename, time }, file) => {
+          const fileTime = Number(file.split("_")[1].split(".")[0]);
+          if (time < fileTime) {
+            filename = file;
+            time = fileTime;
+          }
+          return { filename, time };
+        },
+        { filename: null, time: 0 }
+      ).filename;
+
+      res.sendResponse(200, {
+        fullpath: last ? `${req.protocol}://${req.get("host")}/molecules/${last}` : null,
+        shortpath: `/molecules/${last}`,
+        file: last,
+      });
+    })
+    .catch(res.sendServerError);
+}
+
+export default { importMolecules, getLastImportedFile };
+
+// ********* INTERNAL FUNCTIONS *********
+
+/**
+ * Move a file
+ * @param {string} oldPath
+ * @param {string} newPath
+ * @return {Promise}
+ */
+function moveFile(oldPath, newPath) {
+  return new Promise((resolve, reject) => {
+    fs.rename(oldPath, newPath)
+      .then(resolve)
+      .catch((error) => reject(addErrorTitle(error, "Can't move the file")));
+  });
+}
+
+/**
+ * Delete a file
+ * @param {string} filename
+ * @returns {Promise}
+ */
+function deleteFile(filename) {
+  return new Promise((resolve, reject) => {
+    fs.unlink(filename)
+      .then(resolve)
+      .catch((error) => reject(addErrorTitle(error, "Can't delete the file")));
+  });
+}
+
+/**
+ * Create a directory if it does not exist
+ * @param {string} dirname
+ * @return {Promise}
+ */
+function createDir(dirname) {
+  return new Promise((resolve, reject) => {
+    fs.mkdir(dirname)
+      .then(resolve)
+      .catch((error) => {
+        if (error.code === "EEXIST") {
+          resolve();
+        } else {
+          reject(error);
+        }
+      });
+  });
+}
```

Comment:
You could use [`fs.mkdirSync(dirname, { recursive: true })`](https://stackoverflow.com/a/24311711)

Reply:
Good idea, with that we don't need to check the code of the error !